### PR TITLE
Update: Change modeling approach to polymorphic

### PR DESCRIPTION
**Pull request by cythrawll**
## Description

After approaching this, it was pointed out that polymorphic approach would be better here.

## Proposed Changes

- Change approach to format options to polymorphic so each format type can have their own set of options.

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by kevinhinterlong — APPROVED** [review]

File: packages/core/addon/models/csv.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import DeliveryFormatFragment, { TypedDeliveryFormat } from './fragments/delivery-format';
+
+export type CSVFormatConfig = {
+  type: 'csv';
+  options?: {};
+};
+
+export default class CSVFormat extends DeliveryFormatFragment implements TypedDeliveryFormat, CSVFormatConfig {
+  @attr('string')
+  type!: CSVFormatConfig['type'];
```
Comment (jkusa) [inline packages/core/addon/models/csv.ts:15]:
```suggestion
  declare type: CSVFormatConfig['type'];
```

File: packages/core/addon/models/csv.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  @attr({ defaultValue: () => ({}) })
+  options!: CSVFormatConfig['options'];
```
Comment (jkusa) [inline packages/core/addon/models/csv.ts:18]:
```suggestion
  declare options: CSVFormatConfig['options'];
```

File: packages/core/addon/models/gsheet.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+export type GsheetFormatConfig = {
+  type: 'gsheet';
+  options?: {
+    overwriteFile: boolean;
+  };
+};
+
+export default class GsheetFormat extends DeliveryFormatFragment implements TypedDeliveryFormat, GsheetFormatConfig {
+  @attr('string')
+  type!: GsheetFormatConfig['type'];
```
Comment (jkusa) [inline packages/core/addon/models/gsheet.ts:17]:
```suggestion
  declare type: GsheetFormatConfig['type'];
```

File: packages/core/addon/models/gsheet.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  @attr({ defaultValue: () => ({ overwriteFile: false }) })
+  options!: GsheetFormatConfig['options'];
```
Comment (jkusa) [inline packages/core/addon/models/gsheet.ts:20]:
```suggestion
  declare options: GsheetFormatConfig['options'];
```

File: packages/core/addon/models/pdf.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import DeliveryFormatFragment, { TypedDeliveryFormat } from './fragments/delivery-format';
+
+export type PDFFormatConfig = {
+  type: 'pdf';
+  options?: {};
+};
+
+export default class PDFFormat extends DeliveryFormatFragment implements TypedDeliveryFormat, PDFFormatConfig {
+  @attr('string')
+  type!: PDFFormatConfig['type'];
```
Comment (jkusa) [inline packages/core/addon/models/pdf.ts:15]:
```suggestion
  declare type: PDFFormatConfig['type'];
```

File: packages/core/addon/models/pdf.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+  @attr({ defaultValue: () => ({}) })
+  options!: PDFFormatConfig['options'];
```
Comment (jkusa) [inline packages/core/addon/models/pdf.ts:18]:
```suggestion
  declare options: PDFFormatConfig['options'];
```

File: packages/core/addon/models/png.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import DeliveryFormatFragment, { TypedDeliveryFormat } from './fragments/delivery-format';
+
+export type PNGFormatConfig = {
+  type: 'png';
+  options?: {};
+};
+
+export default class PNGFormat extends DeliveryFormatFragment implements TypedDeliveryFormat, PNGFormatConfig {
+  @attr('string')
+  type!: PNGFormatConfig['type'];
```
Comment (jkusa) [inline packages/core/addon/models/png.ts:15]:
```suggestion
  declare type: PNGFormatConfig['type'];
```

File: packages/core/addon/models/png.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+  @attr({ defaultValue: () => ({}) })
+  options!: PNGFormatConfig['options'];
```
Comment (jkusa) [inline packages/core/addon/models/png.ts:18]:
```suggestion
  declare  options: PNGFormatConfig['options'];
```

File: packages/core/addon/models/csv.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
+/**
```
Comment (jkusa) [inline packages/core/addon/models/csv.ts:1]: Can we move these fragments under `packages/core/addon/models/fragments/delivery-formats/`

**Review by jkusa — COMMENTED** [review]
A few small things, otherwise looks good 👍

File: packages/core/addon/models/csv.ts (hunk 9 above)
Comment (cythrawll) [reply packages/core/addon/models/csv.ts:1]: no because it can't find the schema. even if you register it under a different 'name'. I tried

**Review by jkusa — APPROVED** [review]

**Pull request merged**